src/components/shared/Form/Update/Profile/UpdateProfileForm.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+import React, { useEffect, useState } from "react";
+
+import {
+  Step1,
+  Step2,
+  Step3,
+  Step4,
+} from "@/components/shared/Form/Onboarding/Steps/Steps";
+
+import { Session } from "next-auth";
+import { Flip, toast } from "react-toastify";
+import { useForm } from "react-hook-form";
+import { User } from "~/models/user";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { OnboardingFormSchema } from "~/lib/validators/onboarding.schema";
+import { updateUser } from "~/lib/actions/users";
+import { useRouter } from "next/navigation";
+import router from "next/router";
+
+interface UpdateProfileFormProps {
+  session: Session | null;
+  userFromDB: User;
+}
+
+const UpdateProfileForm: React.FC<UpdateProfileFormProps> = ({
+  session,
+  userFromDB,
+}) => {
+  const router = useRouter();
+
+  //cleaning up the data to load to the form
+  const learningGoalsToPass = Object.entries(userFromDB?.learningGoals).map(
+    (goal) => {
+      return { value: goal[1].goal, completed: goal[1].done };
+    }
+  );
+
+  const knowledgeLevelsToPass = Object.entries(userFromDB?.knowledgeLevels).map(
+    (item) => {
+      return { value: item[1] };
+    }
+  );
```

Comment:
```suggestion
  const knowledgeLevelsToPass = (userFromDB?.knowledgeLevels).map((item) => {
    return { value: item };
  });
```
Same thing here, we can simplify the way we're mapping the data if the data is already an array